pipeline/extract.py

```diff
@@ -49,101 +47,50 @@ def s3_connection() -> client:
         return None
     return s3
 
-def extract_bluesky_files(s3: client) -> list[str]:
-    """Accesses files from S3 and returns a list of texts with the topic present"""
-    continuation_token = None
-    file_names = []
-
-    bucket_parameters = {'Bucket': os.environ.get("S3_BUCKET_NAME")}
-    if not bucket_parameters.get('Bucket'):
-        raise KeyError("Missing environment variable: S3_BUCKET_NAME")
-
-    while True:
-        if continuation_token:
-            bucket_parameters['ContinuationToken'] = continuation_token
 
-        response = s3.list_objects_v2(**bucket_parameters)
-        contents = response.get("Contents")
-
-        if not contents:
-            break
-
-        for file in contents:
-            if file['Key'].endswith(".txt"):
-                file_names.append(file['Key'])
-
-        continuation_token = response.get('NextContinuationToken')
-        if not continuation_token:
-            break
-
-    return file_names
-
-def fetch_file_content(s3: client, file_name: str, topic: list[str]) -> dict:
-    """Checks files from S3 for keywords and returns relevant data if keyword is found"""
-    try:
-        file_obj = s3.get_object(Bucket=os.environ.get("S3_BUCKET_NAME"), Key=file_name)
-        file_content = file_obj['Body'].read().decode('utf-8')
-        hour_folder = file_name.split('/')[-2]  # Extracts the hour folder (e.g., "16/")
+def average_sentiment_analysis(keyword: str, file_data: dict) -> tuple:
+    """Calculates the average sentiment for a keyword in a .json file"""
+    total_sentiment = 0
+    mentions = 0
+    for text, sentiment in file_data.items():
+        if keyword in text:
+            total_sentiment += sentiment['Sentiment Score']['compound']
+            mentions += 1
+    if mentions == 0:
+        return 0, 0
+    return total_sentiment/mentions, mentions
```

Comment:
could the tuple return be:
```suggestion
def average_sentiment_analysis(keyword: str, file_data: dict) -> tuple:
    """Calculates the average sentiment for a keyword in a .json file"""
    total_sentiment = 0
    mentions = 0
    for text, sentiment in file_data.items():
        if keyword in text:
            total_sentiment += sentiment['Sentiment Score']['compound']
            mentions += 1
    if mentions == 0:
        return (total_sentiment, mentions)
    return total_sentiment/mentions, mentions
```